openmetadata-ui/src/main/resources/ui/src/pages/EntityVersionPage/EntityVersionPage.component.tsx

```diff
@@ -60,167 +78,326 @@ const EntityVersionPage: FunctionComponent = () => {
   >([]);
 
   const backHandler = () => {
-    history.push(getDatasetDetailsPath(datasetFQN));
+    switch (entityType) {
+      case EntityType.TABLE:
+        history.push(getDatasetDetailsPath(entityFQN));
+
+        break;
+
+      case EntityType.TOPIC:
+        history.push(getTopicDetailsPath(entityFQN));
+
+        break;
+
+      default:
+        break;
+    }
   };
 
   const versionHandler = (v = version) => {
-    history.push(getDatasetVersionPath(datasetFQN, v as string));
+    history.push(getVersionPath(entityType, entityFQN, v as string));
+  };
+
+  const setBreadCrumbTitle = (
+    value: TitleBreadcrumbProps['titleLinks'] = []
+  ) => {
+    setSlashedTableName(value);
   };
 
   const fetchEntityVersions = () => {
     setIsloading(true);
-    getTableDetailsByFQN(
-      getPartialNameFromFQN(datasetFQN, ['service', 'database', 'table'], '.'),
-      ['owner', 'tags']
-    )
-      .then((res: AxiosResponse) => {
-        const { id, owner, tags, name, database } = res.data;
-        setTier(getTierTags(tags));
-        setOwner(getOwnerFromId(owner?.id));
-        setCurrentVersionData(res.data);
-        getDatabase(database.id).then((resDB: AxiosResponse) => {
-          getServiceById('databaseServices', resDB.data.service?.id).then(
-            (resService: AxiosResponse) => {
-              setSlashedTableName([
-                {
-                  name: resService.data.name,
-                  url: resService.data.name
-                    ? getServiceDetailsPath(
-                        resService.data.name,
-                        resService.data.serviceType,
-                        ServiceCategory.DATABASE_SERVICES
-                      )
-                    : '',
-                  imgSrc: resService.data.serviceType
-                    ? serviceTypeLogo(resService.data.serviceType)
-                    : undefined,
-                },
-                {
-                  name: resDB.data.name,
-                  url: getDatabaseDetailsPath(resDB.data.fullyQualifiedName),
-                },
-                {
-                  name: name,
-                  url: '',
-                  activeTitle: true,
-                },
-              ]);
-            }
-          );
-        });
-        getTableVersions(id)
-          .then((vres: AxiosResponse) => {
-            setVersionList(vres.data);
-            setIsloading(false);
+    switch (entityType) {
```

Comment:
This can be a part of util method

Reply:
@darth-coder00 Here we are making API calls based on the entity type and based on that we are setting some states.